frontend/src/lib/k8s/api/v2/hooks.ts

```diff
@@ -179,225 +183,22 @@ const getWorkingEndpoint = async (endpoints: KubeObjectEndpoint[], cluster: stri
  *
  * @params endpoints - List of possible endpoints
  */
-const useEndpoints = (endpoints: KubeObjectEndpoint[], cluster: string) => {
+export const useEndpoints = (
+  endpoints: KubeObjectEndpoint[],
+  cluster?: string,
+  namespace?: string
+) => {
   const { data: endpoint } = useQuery({
-    enabled: endpoints.length > 1,
+    enabled: endpoints.length > 1 && cluster !== undefined,
```

Comment:
Have you considered the cluster == '' case?

Reply:
yes, empty string is a valid cluster value